plugins/modules/network_compliance_workflow_manager.py

```diff
@@ -810,10 +831,26 @@ def get_want(self, config):
                 self.module.fail_json(msg)
 
             compliance_details = self.modify_compliance_response(response, mgmt_ip_instance_id_map)
-            required, msg = self.is_sync_required(compliance_details, mgmt_ip_instance_id_map)
+            required, self.msg, categorized_devices = self.is_sync_required(compliance_details, mgmt_ip_instance_id_map)
+            self.log("Is Sync Requied: {0} {1}".format(required, self.msg), "DEBUG")
             if not required:
-                self.log(msg, "ERROR")
-                self.module.fail_json(msg)
+                self.update_result("success", False, self.msg, "INFO")
+                self.module.exit_json(**self.result)
+
+            # Get the device IDs of devices in the "OTHER" category and "COMPLIANT" category
+            other_device_ids = categorized_devices.get("OTHER", {}).keys()
+            compliant_device_ids = categorized_devices.get("COMPLIANT", {}).keys()
+            excluded_device_ids = set(other_device_ids) | set(compliant_device_ids)
+
+            if excluded_device_ids:
+                # Exclude devices in the "OTHER" category from sync_device_config_params
+                sync_device_config_params["deviceId"] = [device_id for device_id in mgmt_ip_instance_id_map.values() if device_id not in excluded_device_ids]
+                self.log(
+                    "Skipping these devices because their compliance status is not 'NON_COMPLIANT': {0}".format(
+                        categorized_devices.get("OTHER")
```

Comment:
are we not printing the excluded_device_ids? 
        "Skipping these devices because their compliance status is not 'NON_COMPLIANT': {0}".format(
            **excluded_device_ids**)

Reply:
I am not sure .join works. If yes, then we can add comma. Otherwise, you can ignore it.

```
    excluded_devices_log = ', '.join(excluded_device_ids)
    self.log(
        "Skipping these devices because their compliance status is not 'NON_COMPLIANT': {0}".format(excluded_devices_log),
        "WARNING"
    )

```